### ISSUE TITLE
zipFile.getFileHeader(fileName) is Null on Linux, but works on Windows

### ISSUE THREAD

**Issue by IkoRein** [srikanth-lingala/zip4j#312]
Hi,

First thank you for your great work.

Bit odd issue here. When I try to get BufferedReader with the zipFile.getInputStream(fileHeader), it works on Windows, but not on Linux.

Tested with Java 8 and Java 15 on Linux and it fails on both. Using the 2.7.0 branch of zip4j.

Attached is short code to show how it fails. On Windows this works, but on Linux I get this error:
net.lingala.zip4j.exception.ZipException: FileHeader is null, cannot get InputStream
	at net.lingala.zip4j.ZipFile.getInputStream(ZipFile.java:1024)

Any ideas to how to avoid this? Am I adding the files to the .zip in wrong way?

In my real code, on the "header == null" case I am now extracting the file, then getting it as bufferedReader and deleting the file when it has been read, as aim is to only read the file from the .zip and use the data.

==========================
```
public class ZipTest {

    private static char[] pwd = {'a', 'b', 'c'};

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {

        String fileName = "packed.txt";

        /////////////////////
        // Zip the file
        String zipFileName = "";
        //
        File file = new File(fileName.toLowerCase());
        try {
            // 
            ZipParameters zipParameters = new ZipParameters();
            zipParameters.setEncryptFiles(true);
            zipParameters.setEncryptionMethod(EncryptionMethod.AES);
            zipParameters.setAesKeyStrength(AesKeyStrength.KEY_STRENGTH_256);

            List<File> filesToAdd = Arrays.asList(
                    file
            );

            ZipFile zipFile = new ZipFile(file.getAbsoluteFile() + "z", pwd);
            zipFileName = file.getAbsoluteFile() + "z";
            zipFile.setCharset(StandardCharsets.UTF_8);

            System.out.println(" to " + zipFile.getFile().getAbsolutePath() + "");
            zipFile.addFiles(filesToAdd, zipParameters);

        } catch (Exception ex) {
            ex.printStackTrace(System.out);
        }
        
        /////////////////////////////////
        // Just try to extract the zipped file, works on all platforms
        try {
            new ZipFile(zipFileName,pwd).extractAll("destination_directory");
        } catch (Exception ex) {
            ex.printStackTrace(System.out);
        }

        //////////////////////////
        // Get bufferedreader on the zipped file, fails to get header on linux 
        try {

            // Zipfile 
            ZipFile zipFile = new ZipFile(zipFileName.toLowerCase());
            // password            
            zipFile.setPassword(pwd);
            // set the charset justin case
            zipFile.setCharset(StandardCharsets.UTF_8);

            // File header is the file we are looking for
            // Just test, what file headers we are gettings
             //
           // on Windows gives filenames, on Linux this is empty
            for(FileHeader f: zipFile.getFileHeaders()) {
                System.out.println(f.getFileName());
            }
           ////////////////////////////
            // THIS IS null in Linux ...
            FileHeader fileHeader = zipFile.getFileHeader(fileName);

            // just test
            System.out.println(java.time.LocalTime.now() + " " + ": " + "Getting file " + fileName + " from " + zipFile.getFile().getName());
       
            // create input stream
            ZipInputStream inputStream = zipFile.getInputStream(fileHeader);
            // assign it to the bufferedreader.
            BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
            String line;
            while ((line = br.readLine()) != null) {
                System.out.println(line);
            }
            br.close();

        } catch (IOException ex) {
            ex.printStackTrace(System.out);
        }
    }
}
```
==========================
Thank you for your help.

**Comment by srikanth-lingala** [comment]
Hi, thanks for the detailed explanation, and also the sample code. Helps a lot in analysing the issue.

I will try the sample code and get back.

**Comment by srikanth-lingala** [comment]
I do not have access to a linux machine unfortunately,  but I tried your code on a mac, and it works fine too as long as the `packed.txt` is at the root folder of the project where I run the code from. Was the `packed.txt` also at the root of your project (or in classpath)? Because if it is somewhere else ( for example: `/usr/home/Downloads/paced.txt`, adding this file will work but getting the file header will return a null because the file name in the zip is `paced.txt`, and when you do a `getFileHeader()` you have `/usr/home/Downloads/paced.txt` which will be null because this is not the name of the file in the zip.

**Comment by IkoRein** [comment]
Thank you for looking into this. 

After your feedback I looked at the code and actually it is "my mistake".

On the first unzip part I am using the file as it is, but on the second one I do 'toLowercase()' the filename. And of course this works on Windows, but on Linux/Mac this means, if there are upper case letters in the path, the ZipFile fails.

So the was a bug between the chair and the keyboard. 

Luckily I also found a source for potential multi-platform issue in my own code, which I can now hopefully weed out.
